### PR TITLE
improve stream testing script

### PR DESCRIPTION
**Pull request by MFTabriz**
patch #2525

### CODE REVIEW

File: test/automated/api/run.sh [hunk 1]
```diff
@@ -6,14 +6,13 @@ TEMP_DB=$(mktemp)
...
 # Download a specific version of ffmpeg
-if [ ! -d "ffmpeg" ]; then
-	mkdir ffmpeg
-	pushd ffmpeg >/dev/null || exit
-	curl -sL https://github.com/vot/ffbinaries-prebuilt/releases/download/v4.2.1/ffmpeg-4.2.1-linux-64.zip --output ffmpeg.zip >/dev/null
-	unzip -o ffmpeg.zip >/dev/null
-	PATH=$PATH:$(pwd)
-	popd >/dev/null || exit
-fi
+FFMPEG_PATH=$(mktemp -d)
```
Comment (gabek) [inline test/automated/api/run.sh]: Does this mean it's now downloading a new copy of ffmpeg to a new place every time the tests run? That's a bit excessive. I often run the tests many times and I'd hate to waste all that time and space downloading the same copy of ffmpeg over and over again.

File: test/automated/browser/run.sh [hunk 2]
```diff
@@ -22,15 +22,13 @@ npm install --quiet --no-progress
...
-if [ ! -d "ffmpeg" ]; then
-	echo "Downloading ffmpeg..."
-	mkdir -p /tmp/ffmpeg
-	pushd /tmp/ffmpeg >/dev/null
-	curl -sL --fail https://github.com/vot/ffbinaries-prebuilt/releases/download/v4.2.1/ffmpeg-4.2.1-linux-64.zip --output ffmpeg.zip
-	unzip -o ffmpeg.zip >/dev/null
-	PATH=$PATH:$(pwd)
-	popd >/dev/null
-fi
+FFMPEG_PATH=$(mktemp -d)
```
Comment (gabek) [inline test/automated/browser/run.sh]: Same here re: downloading ffmpeg.

File: test/ocTestStream.sh [hunk 3]
```diff
@@ -1,55 +1,116 @@
-#!/usr/bin/env bash
+#!/bin/bash
 
-# A recent version of ffmpeg is required for the loop of the provided videos
-# to repeat indefinitely.
-# Example: ./test/ocTestStream.sh ~/Downloads/*.mp4 rtmp://localhost/live/abc123
+# Requirements:
+#   ffmpeg (a recent version with loop video support)
```
Comment (gabek) [inline test/ocTestStream.sh:4]: I didn't catch this before, but I thought one of the primary issues we were trying to resolve was people not having ffmpeg on their machine and not understanding how to install it. Can the script download ffmpeg (if not found) just like the other test scripts do? (Assuming it's run on Linux I guess).

Comment (MFTabriz) [reply test/ocTestStream.sh:4]: Our automated test scripts are run in a well-known environments (mostly Linux containers). If we want this script to be used by users who don’t know how to install a package on their own machine, we need to write a fool-proof solution which works everywhere. I don’t know a ready-made solution  for it. Also consider that this script runs on windows too (with cygwin).
What about pointing people to ffmpeg website with a note on how to find the right file instead?

Comment (gabek) [reply test/ocTestStream.sh:4]: Isn't a fool-proof solution to help people work on Owncast what the original goal of https://github.com/owncast/owncast/issues/2502 was?

File: test/automated/api/run.sh (hunk 1 above)
Comment (MFTabriz) [reply test/automated/api/run.sh]: I moved the ffmpeg to a sub-directory. Now, it is only downloaded once.

File: test/automated/browser/run.sh (hunk 2 above)
Comment (MFTabriz) [reply test/automated/browser/run.sh]: fixed here too!

File: test/ocTestStream.sh (hunk 3 above)
Comment (MFTabriz) [reply test/ocTestStream.sh:4]: In that case, I guess this PR is not addressing the issue! 😅 I’ll leave installing ffmpeg to people who are experienced in multi-platform scripting.

**Pull request merged**